### PR TITLE
EDX-2001 Convert StartingAddress attribute between 0-based and 1-based

### PR DESCRIPTION
**Pull request by weichou1229**
Convert StartingAddress attribute between 0-based and 1-based for Modbus device profile

### CODE REVIEW

**Review by FelixTing — APPROVED** [review]
LGTM

File: v1models/transform.go [hunk 1]
```diff
@@ -408,3 +420,37 @@ func toV2ResourceOperations(readWrite string, v1ros []ResourceOperation) []model
 	}
 	return v2ros
 }
+
+// convertStartingAddressToOneBased convert startingAddress attribute from zero-based to one-based when transforming from v2 to v1 profile
+// TODO Remove this func when the Modbus connector supports zero-based startingAddress
```
Comment (cloudxxx8) [inline v1models/transform.go:425]: it's not a TODO anymore
you should only remove this call in xrt bridge when xrt is upgraded

File: v1models/transform.go [hunk 2]
```diff
@@ -33,6 +35,11 @@ func TransformProfileFromV2ToV1(profile models.DeviceProfile) (DeviceProfile, er
 	v1dp.DeviceCommands = toV1DeviceCommands(profile.DeviceCommands)
 	v1dp.CoreCommands = toV1CoreCommand(profile.DeviceResources, profile.DeviceCommands)
 
+	v1dp, err = convertStartingAddressToOneBased(v1dp)
```
Comment (cloudxxx8) [inline v1models/transform.go:38]: retrieve the first Device Resource, you only need to call this function if the attributes contain both "primaryTable" and "startingAddress"

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v1models/transform.go (hunk 1 above)
Comment (weichou1229) [reply v1models/transform.go:438]: I already removed this call from xrt bridge https://github.com/IOTechSystems/xrt-xpert-bridge/pull/9

Comment (cloudxxx8) [reply v1models/transform.go:438]: I mean to remove `\\TODO` here
event if the XRT made change, we will still keep this function for all the v1 Modbus Device Profile

File: v1models/transform.go [hunk 3]
```diff
@@ -33,6 +38,13 @@ func TransformProfileFromV2ToV1(profile models.DeviceProfile) (DeviceProfile, er
 	v1dp.DeviceCommands = toV1DeviceCommands(profile.DeviceCommands)
 	v1dp.CoreCommands = toV1CoreCommand(profile.DeviceResources, profile.DeviceCommands)
 
+	if isModbusAttributes(v1dp.DeviceResources[0].Attributes) {
```
Comment (cloudxxx8) [inline v1models/transform.go:41]: don't hardcode the index here.  it would cause nil pointer panic if DeviceResources is an empty array

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v1models/transform.go (hunk 3 above)
Comment (weichou1229) [reply v1models/transform.go]: Modified. Added code to check slice lenght.

File: v1models/transform.go (hunk 1 above)
Comment (weichou1229) [reply v1models/transform.go]: Got it. Removed the TODO comment and keep the func call in the xrt bridge.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**